doc/_ext/apilinks.py

```diff
@@ -0,0 +1,80 @@
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab.
+
+'''
+This module defines a Sphinx plugin which enables the use of a
+':ref_guide:' role within Sphinx documentation. The link targets
+are then generated using the 'ref_guide_base' (configuration
+variable) as a base. This is set in the 'conf.py' of the documentation
+and enables link targets to be configured at build time.
+
+'''
+from docutils import nodes, utils
+
+
+def setup(app):
+    '''
+    Sets-up this Sphinx plugin.
+
+    :param app: the Sphinx application object.
+    :type app: :py:class:`sphinx.application.Sphinx`
+
+    '''
+    def ref_link_role(role, rawtext, text, lineno, inliner, options={},
+                      content=[]):
+        '''
+        Handler routine called when the ':ref_guide:' role is encountered.
+
+        Modifies the supplied link text by pre-pending the text stored in
+        the 'ref_guide_base' Sphinx configuration variable.
+
+        '''
+        items = text.split()
+        ref = app.config.ref_guide_base + items[-1]
+        if len(items) > 1:
+            # The link had a name associated with it so reconstruct it.
+            name = ' '.join(items[:-1])
+        else:
+            name = utils.unescape(ref)
+        # Construct the new reference.
+        node = nodes.reference(rawtext, name, refuri=ref, **options)
+        return [node], []
+
+    # Define a new Sphinx configuration variable that will be used to
+    # store the base URL to use when generating links.
+    app.add_config_value('ref_guide_base', 'http://localhost/', False)
```

Comment:
No you're right and it could potentially be confusing. I've changed this so that we no longer set a default value - that way the values are only ever set in the `conf.py` file of the document.